ipq40xx: meraki: convert to nvmem for calibration

halmartin opened the pull request:
This commit changes the Meraki MR33 and MR74 device trees to use nvmem for ART calibration.

There is also some device tree clean-up, to align with the work done for the MR30H/Z3/GX20 support.

The WiFi BDF variant was moved from `insect-common.dtsi` to the respective device files in preparation for additional Meraki insect-family devices being added.

Leo-PL commented on target/linux/ipq40xx/files-6.12/arch/arm/boot/dts/qcom/qcom-ipq4029-insect-common.dtsi:
```diff
@@ -330,27 +355,17 @@
-			drive-strength = <16>;
-			bias-disable;
-		};
+		function = "blsp_i2c0";
```
Didn't this slip by in accidentally?
Edit: probably should go to a separate commit for alignment with `wired`-s

Leo-PL commented on target/linux/ipq40xx/files-6.12/arch/arm/boot/dts/qcom/qcom-ipq4029-insect-common.dtsi:
```diff
@@ -385,16 +400,14 @@
-	qcom,ath10k-calibration-variant = "Meraki-MR33";
-	nvmem-cells = <&mac_address 2>;
-	nvmem-cell-names = "mac-address";
+	nvmem-cells = <&precal_factory_1000>;
```
If I'm not mistaken, the MAC address-related NVMEM cells should stay together with new caldata, ART blobs do not contain the MACs on `insects`-s

Leo-PL commented on target/linux/ipq40xx/files-6.12/arch/arm/boot/dts/qcom/qcom-ipq4029-insect-common.dtsi:
```diff
@@ -385,16 +400,14 @@
-	qcom,ath10k-calibration-variant = "Meraki-MR33";
-	nvmem-cells = <&mac_address 3>;
-	nvmem-cell-names = "mac-address";
+	nvmem-cells = <&precal_factory_5000>;
```
If I'm not mistaken, the MAC address-related NVMEM cells should stay together with new caldata, ART blobs do not contain the MACs on insects-s

Leo-PL commented on target/linux/ipq40xx/files-6.12/arch/arm/boot/dts/qcom/qcom-ipq4029-insect-common.dtsi:
```diff
@@ -282,8 +307,8 @@
 		reg = <0x00010000 0 0 0 0>;
-		nvmem-cells = <&mac_address 1>;
-		nvmem-cell-names = "mac-address";
+		nvmem-cells = <&cal_factory_9000>;
```
If I'm not mistaken, the MAC address-related NVMEM cells should stay together with new caldata, ART blobs do not contain the MACs on insects-s

halmartin committed:
ipq40xx: meraki: convert to nvmem for calibration

This commit changes the Meraki MR33 and MR74 device trees to use nvmem
for ART calibration.

The WiFi BDF was moved from insect-common.dtsi to the respective
device files in preparation for additional insect-family devices being added.

Signed-off-by: Hal Martin <hal.martin@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/20474
Signed-off-by: Robert Marko <robimarko@gmail.com>

robimarko commented:
Thanks! Rebased on top of main and merged!
